src/crimson/osd/osd_operations/client_request.cc

```diff
@@ -219,25 +219,8 @@ ClientRequest::interruptible_future<> ClientRequest::with_pg_process_interruptib
   DEBUGDPP("{}.{}: pg active, entering process[_pg]_op",
 	   *pgref, *this, this_instance_id);
 
-  {
-    /* The following works around two different gcc bugs:
-     *  1. https://gcc.gnu.org/bugzilla/show_bug.cgi?id=101244
-     *     This example isn't preciesly as described in the bug, but it seems
-     *     similar.  It causes the generated code to incorrectly execute
-     *     process_pg_op unconditionally before the predicate.  It seems to be
-     *     fixed in gcc 12.2.1.
```

Comment:
There is another instance at: https://github.com/ceph/ceph/blob/6469b2c59e13529f13e8a27f8ab5f840a159ba51/src/crimson/osd/ops_executer.cc#L871-L876

Reply:
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=101244 doesn't seem to be fixed yet